examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/MigrationTest.kt

```diff
@@ -0,0 +1,217 @@
+package com.mongodb.realm.realmkmmapp
+
+import io.realm.kotlin.Realm
+import io.realm.kotlin.RealmConfiguration
+import io.realm.kotlin.internal.platform.runBlocking
+import io.realm.kotlin.migration.AutomaticSchemaMigration
+import io.realm.kotlin.types.RealmObject
+import io.realm.kotlin.types.annotations.PrimaryKey
+import org.mongodb.kbson.BsonObjectId
+import org.mongodb.kbson.ObjectId
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import co.touchlab.kermit.Kermit
+import com.mongodb.realm.realmkmmapp.Log.kermit
+import io.realm.kotlin.dynamic.DynamicMutableRealmObject
+import io.realm.kotlin.dynamic.DynamicRealmObject
+import io.realm.kotlin.dynamic.getValue
+import io.realm.kotlin.ext.backlinks
+import io.realm.kotlin.query.RealmResults
+import io.realm.kotlin.types.RealmList
+
+// :replace-start: {
+//   "terms": {
+//     "MigrationExampleV1_": "",
+//     "MigrationExampleV2_": "",
+//     "MigrationExampleV3_": "",
+//     "MigrationExampleV4_": "",
+//     "MigrationExampleV5_": "",
+//     "MigrationExampleV1Update1_": "",
+//     "MigrationExampleV1Update2_": ""
+//   }
+// }
+
+// :snippet-start: model-v1
+// Realm schema version 1
+class MigrationExampleV1_Person : RealmObject {
+    var firstName: String = ""
+    var lastName: String = ""
+    var age: Int = 0
+}
+// :snippet-end:
+// :snippet-start: model-v2
+// Realm schema version 2
+class MigrationExampleV2_Person : RealmObject {
+    var firstName: String = ""
+    var lastName: String = ""
+    var age: Int = 0
+    var email: String? = null
+}
+// :snippet-end:
+// :snippet-start: model-v3
+// Realm schema version 3
+class MigrationExampleV3_Person : RealmObject {
+    var firstName: String = ""
+    var lastName: String = ""
+    // var age: Int = 0
+    var email: String? = null
+}
+// :snippet-end:
+// :snippet-start: model-v4
+// Realm schema version 1 (oldObject)
+class MigrationExampleV4_Person : RealmObject {
+    var _id: ObjectId = ObjectId()
+    var firstName: String = ""
+    var lastName: String = ""
+    var age: Int = 0
+}
+
+// Realm schema version 2 (newObject)
+class MigrationExampleV5_Person : RealmObject {
+    var _id: String = "" // change property type
+    var fullName: String = "" // merge firstName and lastName properties
+    var yearsSinceBirth: Int = 0 // rename property
+}
+// :snippet-end:
+
+class MigrationTest {
+
+    @Test
+    fun updateSchemaVersion1Test() {
+        runBlocking {
+// :snippet-start: update-schema-version-2
+            val config = RealmConfiguration.Builder(
```

Comment:
should these comments be tabbed over to align with the following line?
that's how it works for most languages, i believe.

```suggestion
            // :snippet-start: update-schema-version-2
            val config = RealmConfiguration.Builder(
```

Reply:
this is honestly so I can easily see where my snippets start and end 😅